### PR TITLE
Instrument Spring SessionAwareMessageListener

### PR DESCRIPTION
**Pull request by bantonsson**

### CODE REVIEW

File: dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    group = 'org.springframework'
+    module = 'spring-jms'
+    versions = "[3.1.0.RELEASE,]"
+    extraDependency 'javax.jms:jms-api:1.1-rev-1'
+  }
+}
+
+apply from: "$rootDir/gradle/java.gradle"
+
+dependencies {
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: I think you need latest dependency expressions here

Comment (bantonsson) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: Could you explain why, and how it's supposed to be used? I thought that is meant to run the test on the latest deps, and that doesn't seem to be done in all instrumentations.

Comment (richardstartin) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: The idea is that the library can change and the instrumentation will stop working if it does. Downloading the latest dependency gives an early warning that a library breaking the instrumentation has been released, and that the instrumentation should be amended, or its applicability should be capped via muzzle, and a new instrumentation targeting the new library version should be implemented.

Comment (bantonsson) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: Yes, but I thought that `muzzle` was supposed to catch changes in APIs, which is what we are instrumenting here.

Anyway, I can add a `latestDeps` test section as well. Some other projects reuse the `test` directory for that, is that ok to do?

Comment (richardstartin) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: Muzzle verifies that classes and methods we expect to be present are present, it doesn't run any tests though.

Comment (richardstartin) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: What I meant was add some dependencies to the build file like this from e.g. spring-webflux

```
  latestDepTestCompile group: 'org.springframework.boot', name: 'spring-boot-starter-webflux', version: '2.0.+'
  latestDepTestCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: '2.0.+'
  latestDepTestCompile group: 'org.springframework.boot', name: 'spring-boot-starter-reactor-netty', version: '2.0.+'
```

Comment (bantonsson) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: Is that enough? I thought that we needed a new `testSet` with the `org.unbroken-dome.test-sets` plugin.

Comment (richardstartin) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:12]: Yes that's necessary, as in the webflux example

Comment (bantonsson) [reply dd-java-agent/instrumentation/spring-jms-3.1/spring-jms-3.1.gradle:20]: A `latestDepTest` have been added...

**Review by richardstartin — APPROVED** [review]

**Pull request merged**